pgcli/pgcompleter.py

```diff
@@ -308,15 +307,33 @@ def get_column_matches(self, suggestion, word_before_cursor):
         tables = suggestion.tables
         _logger.debug("Completion column scope: %r", tables)
         scoped_cols = self.populate_scoped_cols(tables)
-
+        colit = scoped_cols.items()
+        flat_cols = []
+        for cols in scoped_cols.values():
+            flat_cols.extend(cols)
         if suggestion.drop_unique:
             # drop_unique is used for 'tb11 JOIN tbl2 USING (...' which should
             # suggest only columns that appear in more than one table
-            scoped_cols = [col for (col, count)
-                                 in Counter(scoped_cols).items()
+            flat_cols = [col for (col, count)
+                                 in Counter(flat_cols).items()
                                    if count > 1 and col != '*']
-
-        return self.find_matches(word_before_cursor, scoped_cols, meta='column')
+        lastword = last_word(word_before_cursor, include='most_punctuations')
+        if lastword == '*':
+            if (lastword != word_before_cursor and len(tables) == 1
+              and word_before_cursor[-len(lastword) - 1] == '.'):
+                # User typed x.*; replicate "x." for all columns
+                sep = ', ' + self.escape_name(tables[0].ref) + '.'
+                collist = (sep).join([c for c in flat_cols if c != '*'])
```

Comment:
This must be incorrect because for columns `foo, bar, baz` you get `foo, x.bar, x.baz`, i.e. it forgets the leading qualifier. I think you want
```
qual = self.escape_name(tables[0].ref) + '.'
collist = ', '.join(qual + c for c in flat_cols if c ~= '*')
```
Note also you don't need parens around the string argument to `join`, e.g. `sep.join(` not `(sep).join(`. You also don't need square brackets inside the join call, e.g. `sep.join(c for c in cols)` not `sep.join([c for c in cols])`